pandas/core/generic.py

```diff
@@ -5906,6 +5908,129 @@ def infer_objects(self: FrameOrSeries) -> FrameOrSeries:
             )
         ).__finalize__(self)
 
+    # ----------------------------------------------------------------------
+    # Convert to types that support pd.NA
+
+    def _as_nullable_dtype(self: ABCSeries) -> ABCSeries:
+        """
+        Handle one Series
+
+        Rules:
+        If an object, see if we can infer string, boolean or integer, otherwise leave
+        alone
+        If an integer and not an extension type, convert to the Int64/Int32 type
+        (platform dependent)
+        If numeric, see if we can infer integer, otherwise try to use astype() to make
+        it integer.
+
+        """
+        dtype = self.dtype
+        new_dtype = dtype
+        changeit = False
+        constructit = True
+        result = self
+        target_int_dtype = "Int64"
+
+        if is_object_dtype(dtype):
+            new_dtype = lib.infer_dtype(self)
+            if new_dtype not in {"string", "boolean", "integer"}:
+                new_dtype = dtype
+            else:
+                changeit = True
+        elif is_integer_dtype(dtype):
```

Comment:
This change may warrant an option. It's not clear to me whether this behavior is always desirable.

The motivating use case is IO from a lossy format like CSV. In that case, you've already got NaNs.

Reply:
@TomAugspurger by "this behavior", you mean converting something of type `int64` to `Int64` being something people can control?  Wouldn't it preferable to always make that change?

Or are you suggesting that we have an option like `keep_integer=False` which, if set to `True` means that we don't change existing integer types at all?

Reply:
Yes, I mean `int` -> `Int`. I think having an option for that would be desirable. I'm not sure what the default would be.

Reply:
added keep_integer